### DISCUSSION TITLE
速度300ms不合适

### DISCUSSION THREAD

**Discussion by mikumifa** [mikumifa/cppTickerBuy#28] [通知]
cpp的风控很强，300ms抢票会封ip，解决方法，就是换电脑的ip （开梯子，换网络
开梯子会导致程序没办法运行，这个是可以通过配置你的梯子软件解决的
建议抢票速度控制在1000ms延迟左右

**Comment by mikumifa** [comment]
风控输出： #17 （这种输出说明你ip被封禁了
开梯子时候出问题： https://blog.csdn.net/petterxf/article/details/139869556

**Comment by Azrael070** [comment]
如果在没法换ip的情况下，将延迟设为1000ms，以尽可能的保障不进入风控？

**Comment by Yusibr** [comment]
并不像是完全的ip封禁策略，感觉是防刷策略，在延迟时间为1500ms的时候，仍旧会出现请求过于频繁的错误，但是过一会后续几次调用也会出现成功的请求，如果是ip封禁的话短时间内应该都调不通的
![$5DF@DX5C~L6QVO~JN3VD)M](https://github.com/user-attachments/assets/d105c27a-fc35-40e6-868c-d174d1d7607e)

**Reply by mikumifa** [reply]
那这样是不是要一直暴力刷新就好了

**Reply by Yusibr** [reply]
我看到源码里使用的是buyTicketWeixin的url，是否可以考虑增加支付宝的buyTicketAlipay渠道来提高成功率，如果是接口防刷策略，可能是针对单一接口的防刷，可以考虑同时请求支付宝渠道和微信渠道，你的签名每次都是重新生成的对吧，我的python不太行

**Reply by mikumifa** [reply]
有道理，可以考虑

**Reply by Yusibr** [reply]
我的猜测是错的，我手动先请求微信渠道接口后立马请求支付宝渠道接口，微信提示未到开票时间但是支付宝渠道被拒绝了，提示请求过于频繁，看来是可能基于ip的防刷策略而不是针对单独接口的，并且在手动操作下也会出现请求过于频繁，我间隔5s的请求仍旧被拒绝，感觉是支付接口上很暴力基于单位时间内不允许重复请求的策略

**Reply by qiyezhishang** [reply]
![QQ_1727603117201.png](https://github.com/user-attachments/assets/5e9ce68f-95a2-4509-8d0a-13609320c468)

控制到10秒就一直是未到时间了😨但是这样会不会太慢，这样是不是只能用多台设备同时抢了啊（啥也不懂我瞎说的）

**Reply by Yusibr** [reply]
@qiyezhishang 从目前看来针对平民（没错就是我们），最好的办法是你使用多ip（你有多台PC或者你有很多正向代理的渠道），同时抢加上手机辅助了

**Reply by lwei65** [reply]
@qiyezhishang 多PC可以使用同一个WiFi吗

**Reply by nighttosakura** [reply]
> @qiyezhishang 多PC可以使用同一个WiFi吗

不能，都是走的运营商给你的动态ip
